Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp

```diff
@@ -355,7 +356,28 @@ namespace AssetProcessor
             AssetProcessor::AssetDatabaseConnection assetDatabaseConnection;
             assetDatabaseConnection.OpenDatabase();
 
-            auto jobsFunction = [this, &assetDatabaseConnection](AzToolsFramework::AssetDatabase::JobDatabaseEntry& entry)
+            // Get historical ProcessJob stats from asset database
+            AZStd::unordered_map<QueueElementID, AZ::s64> historicalStats;
+            auto statsFunction = [&historicalStats](AzToolsFramework::AssetDatabase::StatDatabaseEntry entry)
+            {
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(entry.m_statName, tokens, ',');
+
+                if (tokens.size() == 4)
+                {
+                    QueueElementID elementId;
+                    elementId.SetInputAssetName(tokens[1].c_str());
+                    elementId.SetJobDescriptor(tokens[2].c_str());
+                    elementId.SetPlatform(tokens[3].c_str());
+                    historicalStats[elementId] = entry.m_statValue;
+                }
```

Comment:
My initial thought was that, since statName can be arbitrary, statName starting with `ProcessJob,` but does not have 4 in token size is not necessarily an error, so we simply ignore it. 

However, since it is only us (Asset Processor) using this StatsCapture, we should have full control over what StatNames are. It sounds reasonable to make a warning here.

Reply:
Updated. Emit warning message if token count is not 4. 
Users cannot capture stats, so this issue could happen only if 
1) some Asset Processor code accidentally uses `StatsCapture` that has `ProcessJob,` as its name's prefix, but is not our metric code, or
2) database is corrupted.

Both cases are rare. In the former case, I think users can do little but create an issue so that Asset Processor developers can inspect. In the latter case, none of the database should be trusted. Probably the user should be prompt to drop the database at a global (higher) level.